app.js

```diff
@@ -20,6 +21,53 @@ const auth = authorization(app);
 app.use(auth.initialize());
 app.auth = auth;
 
+let detailsRoute = function(req, res, next) {
+    console.log(req.method + " " + req.url);
+    
+    if (req.headers && req.headers.authorization) {
+        console.log("Token: " + req.headers.authorization.substring(4, req.headers.authorization.lenght));
+    }
```

Comment:
Is this working?, 
Because`lenght` doesn't exist 

```Javascript
req.headers.authorization.lenght
```

Reply:
Hi Vinny,

Yes works. Because this is a string. So firstly I check if this header exist and after I check this. I know that this variable could have different informations, but always arrive like string correct? Or this is not good practice.

Reply:
it doesn't work because you misspelled `length`.
Let's do something more simple. Remove this middleware that you created.

What I wanted you to do was just to `console.log()` the req that is coming and to create a new value in `headers`. 

Lets keep the code simple because I am going to ask you to right tests for this function.

```Javascript
(req, res, next) => { 
  console.log('this is the request', req.body);
   req.headers['new-header'] = 'hello'
    next();
}
```

That was pretty much what I asked you to do, so reset your code and add this middleware to your app.